runners/flink/src/main/java/org/apache/beam/runners/flink/FlinkBatchPortablePipelineTranslator.java

```diff
@@ -254,15 +270,22 @@ public void translate(
             stagePayload.getEnvironment(),
             PipelineOptionsTranslation.toProto(context.getPipelineOptions()),
             outputMap);
-    Map<String, String> inputs = transform.getInputsMap();
-    DataSet<WindowedValue<InputT>> inputDataSet =
-        context.getDataSetOrThrow(Iterables.getOnlyElement(inputs.values()));
-    DataSet<RawUnionValue> taggedDataset =
+
+    DataSet<WindowedValue<InputT>> inputDataSet = context.getDataSetOrThrow(stagePayload.getInput());
+
+    MapPartitionOperator<WindowedValue<InputT>, RawUnionValue> taggedDataset =
         new MapPartitionOperator<>(inputDataSet,
             typeInformation,
             function,
             transform.getUniqueName());
 
+    for (Map.Entry<String, String> sideInput : stagePayload.getSideInputsMap().entrySet()) {
+      // register under the global PCollection name, only ExecutableStageFunction needs to
+      // know the mapping from local name to global name
```

Comment:
Does this mean that it's sufficient to use the local name of the consumer and the global collection id without also storing the PTransform that references it? How do we know from the runner side what type of view the transform wants?